Remove codegen logic to support non-c10-full ops

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49613 Remove backward and requires_grad from Autograd backend key
* #49259 Remove `use_c10_dispatcher: full` lines
* #49251 Remove generated_unboxing_wrappers and setManuallyBoxedKernel
* #49220 Remove .impl_UNBOXED() and functionalities associated with it
* **#49164 Remove codegen logic to support non-c10-full ops**

This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

smessmer committed:
Remove codegen logic to support non-c10-full ops

This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]

bhosmer commented on tools/autograd/gen_variable_type.py:
```diff
@@ -887,16 +862,12 @@ def unpack_args(f: NativeFunction) -> Tuple[List[str], List[Binding]]:
-        bindings = list(sig_group.signature.arguments())
+    bindings = [r for a in f.func.schema_order_arguments()
+                for r in cpp.argument(a,
+                                        method=False,
```
total nit but if you're in here again before landing, it'd be nice to indent these in the conventional way

bhosmer reviewed (approved):
Such a satisfying stack 😁

smessmer committed:
Update on "Remove codegen logic to support non-c10-full ops"


This PR removes the logic paths in codegen that were responsible for handling non-c10-full ops.
This only goes through our basic codegen. It does not simplify C++ code yet and it does not remove the codegen for generated unboxing wrappers yet.

Differential Revision: [D25462977](https://our.internmc.facebook.com/intern/diff/D25462977/)

[ghstack-poisoned]
